### PR TITLE
WIP Add compatibility with channels_last_3d

### PR DESCRIPTION
**Pull request by dmenig**
Maybe satisfies a good chunk of #59168
cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: android/pytorch_android/src/androidTest/java/org/pytorch/PytorchTestBase.java [hunk 1]
```diff
@@ -367,16 +384,106 @@ public void testChannelsLastConv2d() throws IOException {
...
         MemoryFormat.CHANNELS_LAST,
         new long[] {1, 3, 2, 2},
-        new long[] {2, 11, -101, 4, 12, -102, 6, 13, -103, 8, 14, -104});
-  }
+        new long[] {
+          2*111, 211, -311,      2*112, 212, -312,
+          2*121, 221, -321,      2*122, 222, -322});
+  }
+
+  // @Test
```
Comment (mikaylagawarecki) [inline android/pytorch_android/src/androidTest/java/org/pytorch/PytorchTestBase.java:408]: why is this test commented out?

**Review by mikaylagawarecki — CHANGES_REQUESTED** [review]
Hey @hyperfraise, thanks for the contribution! This is a very large PR, could you please break down the changes into separate PRs based on the modules in the linked issue

File: android/pytorch_android/src/androidTest/java/org/pytorch/PytorchTestBase.java (hunk 1 above)
Comment (hyperfraise) [reply android/pytorch_android/src/androidTest/java/org/pytorch/PytorchTestBase.java:408]: I'll split it, thx.

Comment (hyperfraise) [reply android/pytorch_android/src/androidTest/java/org/pytorch/PytorchTestBase.java:408]: Would you please review those ? 
https://github.com/pytorch/pytorch/pull/114790
https://github.com/pytorch/pytorch/pull/114791
https://github.com/pytorch/pytorch/pull/114792
https://github.com/pytorch/pytorch/pull/114793
https://github.com/pytorch/pytorch/pull/114794
https://github.com/pytorch/pytorch/pull/114798

**Pull request closed without merging**